pwa/src/apiService/resources/components.ts

```diff
@@ -37,4 +37,5 @@ const deletes = [
   { name: "resultDisplayLayout" },
   { name: "dependenciesDisplayLayout" },
   { name: "landingDisplayLayout" },
+  { name: "catagoryDisplayLayout" },
```

Comment:
We're currently duplicating this code on four places. Even though this piece of logic is _very_ generic. It doesn't change within the different implementations (nor should it!).

Therefore, I suggest moving this elsewhere, like so (working code, dummy comments):

```
// filtersToQueryParams.ts

// First, we can simplify the current implementation of deleting keys (right now, we need an object to be passed, I think a simple string would do the trick)

Object.keys(filters)
    .filter((key) => filterKeysToRemove.includes(key))
    .forEach((key) => {
      delete filters[key];
    });

// .. other logic of the file

// List of keys that are in IFilters, but should never be sent to the gateway as query params <= KEEP THIS COMMENT :)
const filterKeysToRemove: string[] = [
  "resultDisplayLayout",
  "dependenciesDisplayLayout",
  "landingDisplayLayout",
  "catagoryDisplayLayout",
];
```

Reply:
This way, we only need to keep track of this logic in one file and we can add it a  simple string array.